### PR TITLE
Split Install for Custom ClusterIssuer

### PR DESCRIPTION
**Pull request by manno**
The cert-manager CRD neeeds to exist, before operators of Epinio can install with a custom cluster issuer.

* add `install-cert-manager` to allow for a split install: cert-manager first, then epinio
* explicitly check for custom issuers
* wait for issuer to exists before continuing with install
* add example for private CA to docs

refers to #63 

Test data:

<details>
  <summary>Click to expand</summary>

```
---
apiVersion: v1
kind: Secret
metadata:
  name: private-ca-secret
  namespace: cert-manager
data:
  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUZIekNDQXdlZ0F3SUJBZ0lVUEdHVXVOT0ZOUDRHNXdyR2x4dWFwUkFzQy9zd0RRWUpLb1pJaHZjTkFRRUwKQlFBd0h6RWRNQnNHQTFVRUF3d1VLaTV2YldjdWFHOTNaRzlwTG5kbFluTnBkR1V3SGhjTk1qRXdPREF6TVRVeApNalEzV2hjTk16RXdPREF4TVRVeE1qUTNXakFmTVIwd0d3WURWUVFEREJRcUxtOXRaeTVvYjNka2Iya3VkMlZpCmMybDBaVENDQWlJd0RRWUpLb1pJaHZjTkFRRUJCUUFEZ2dJUEFEQ0NBZ29DZ2dJQkFMbXp0U3g4eFBoQ3BaemkKdUNaVGFBajVkQ3dXS3hnWkFHb2xaQmw0czJQOEcyU1c3YnhUQ2JtSm5udmlxS3VYY3dNMCtjWjc3RWNoajI4SQpwVisyUWlRenFRaTZlb0E4WmQ5Z0piSVdjeUVhdEV3TmFocmdpYkNGb1ovQ040OGhXckF6Ukt3QzB6ZDVCc3pYCnpRaHc5OS9rZGhMdDEzaWlsWUlqcEVoMCtjSGU0NjF2ajVYRUpnS0FKL1NQSVZDWkpOUWVoS1paOXpLb0xJQTQKNkxxMmhJOFpDWHRJdVlOaTh3cXEwRUdnT0owbVk0bjlwaExJRDdJdU4ycmhzVisyb1lQVm9wTzQ0ZkZrOWxJdQo4dEsrUGhJK2ZBSk1DZlJlUTNnaUZZb2FZQTlyUjRid0VpNGJ0QUNwV0xuN3g0TkVlZUdRT0RNd09hVmRBdVpFCjhRM3ZwSENtOEtaRnh4N1RzUDRoQ2F4dXZpcFk4L09SR2FZRlZOeS9KMklhckFwcmVXY2RyYTVoRWp4MDlqR3oKZm43VnpaclNOOTBkaUUzcmFKajRwVmdIR1dwUUNtRi9QRDEvMjRyUXA3N1J3cXdXNHNIUVdiMzJRcjNOOUdDUQpKNi8yc0tHZU1USlJyOUFNZysvRGg3Z0drMnhURXQyTjFSTUJ5SlgxVEliSzlDOEdkd1p3VWJoWDg4RGt5TTBuCkhWd0pIRm5xYTZOejVjNmw1NzcrWHVYYUhIcXN6RjhMUXJUTjI2Y2o5emQralVWcVVrQm9jRkxJR0d3KzVXYzkKMGZRR3U5R1prQ09yenpzbWZ3SUtJdy9haHBIYUdTbDMvUTlwR0ZxeEJMTEo5S1N2dEVrRlJkK0ZsdVdKVVovUQphcjdZN28wd0RWb1NoSy9QbVgvT2xVYmViVkhIQWdNQkFBR2pVekJSTUIwR0ExVWREZ1FXQkJUeGdPSTMxdlBSCngvVjhTQW1EMHV5azAyRFFSVEFmQmdOVkhTTUVHREFXZ0JUeGdPSTMxdlBSeC9WOFNBbUQwdXlrMDJEUVJUQVAKQmdOVkhSTUJBZjhFQlRBREFRSC9NQTBHQ1NxR1NJYjNEUUVCQ3dVQUE0SUNBUUJpVHJMWW84STJqRGh4bURjMQpvUVl0dTdZUEVyWldpMGhIOC92VU82ME5uVEc0V3J0ZGo5V3pIUk9jTjA0RzlJQnd0VEpQQWpiV0piSktSc3Q0Ckc2T3lYTXhCNW80L2V5UVVjc3dEV09GNHZKa0xRbnBiNDI3WWhLZ0o2bCtKRm5KaDRETzFjbGRzZVU5em5COGUKdDIvLzJZYk9uQkJMSUdKNkFyb3RtZEs3YmpWTzcrRmhBbWJ5bGtBVm95eHVKRThlNEtoNjFvR0dnUkpWY05rYgovYlF3SzJKUkp4cEhZM0lXMGsrZGtjNVRmZFpWVXkrWHp4WE83cUV0QndQeFNFNUFUamJjVzdaLzMweGljVWdBClhaOW9BR3Q2SzJIUnNCcUhpTURzcENsbmxLZWVJWEZLS3ZITzNOR3pJNXhYZjM5NmlIQkx6VWRKb3ZCbjkxSlYKMmYyV1hraVZ2QTVCVmtrMXNhaU9YQ2pHRlhVVnlJT0M1aGVJVXJWeWwwMnJqckg2YU0xTUxMZmJUemVuZm5VcQpTaURhWVNOUkUyanZFUDh6aFZZbU9KZ1RLODV5cThkN0JUckozdC94TEpCc09IaW5YVTJOalVUb3N3NnZTSW9wCjlyUXE0OEVNY1BuS2pMMUlSZ3cvZWs5d2FHVjk5ZWx6WUhHUXpteEdkM09aWFp4ZExibWVoYm5yNVpYQ0pnb3QKaUZyU0RzakVuVFdnS1ppYzE5dklDV2t0YnMzN3BQbEl3em1PclNYd2wrb3lHd0xtdHMxWXcvcHMyTDhnc0p4MQpjTnM0ZjBqQkZMUDJIMnRheGRHdEdLamNVN3Y1YWNuakE0bVFod3grTDY5cXE3QWZYemR6NllmT3p1SXZ6eEtsCmJkbjRqVUJ6bW9hd2lwK2p0ZzdxTGdzUllnPT0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
  tls.key: LS0tLS1CRUdJTiBQUklWQVRFIEtFWS0tLS0tCk1JSUpSQUlCQURBTkJna3Foa2lHOXcwQkFRRUZBQVNDQ1M0d2dna3FBZ0VBQW9JQ0FRQzVzN1VzZk1UNFFxV2MKNHJnbVUyZ0krWFFzRmlzWUdRQnFKV1FaZUxOai9CdGtsdTI4VXdtNWlaNTc0cWlybDNNRE5QbkdlK3hISVk5dgpDS1ZmdGtJa002a0l1bnFBUEdYZllDV3lGbk1oR3JSTURXb2E0SW13aGFHZndqZVBJVnF3TTBTc0F0TTNlUWJNCjE4MEljUGZmNUhZUzdkZDRvcFdDSTZSSWRQbkIzdU90YjQrVnhDWUNnQ2YwanlGUW1TVFVIb1NtV2ZjeXFDeUEKT09pNnRvU1BHUWw3U0xtRFl2TUtxdEJCb0RpZEptT0ovYVlTeUEreUxqZHE0YkZmdHFHRDFhS1R1T0h4WlBaUwpMdkxTdmo0U1Bud0NUQW4wWGtONEloV0tHbUFQYTBlRzhCSXVHN1FBcVZpNSs4ZURSSG5oa0Rnek1EbWxYUUxtClJQRU43NlJ3cHZDbVJjY2UwN0QrSVFtc2JyNHFXUFB6a1JtbUJWVGN2eWRpR3F3S2EzbG5IYTJ1WVJJOGRQWXgKczM1KzFjMmEwamZkSFloTjYyaVkrS1ZZQnhscVVBcGhmenc5Zjl1SzBLZSswY0tzRnVMQjBGbTk5a0s5emZSZwprQ2V2OXJDaG5qRXlVYS9RRElQdnc0ZTRCcE5zVXhMZGpkVVRBY2lWOVV5R3l2UXZCbmNHY0ZHNFYvUEE1TWpOCkp4MWNDUnhaNm11amMrWE9wZWUrL2w3bDJoeDZyTXhmQzBLMHpkdW5JL2MzZm8xRmFsSkFhSEJTeUJoc1B1Vm4KUGRIMEJydlJtWkFqcTg4N0puOENDaU1QMm9hUjJoa3BkLzBQYVJoYXNRU3l5ZlNrcjdSSkJVWGZoWmJsaVZHZgowR3ErMk82Tk1BMWFFb1N2ejVsL3pwVkczbTFSeHdJREFRQUJBb0lDQVFDMW00WG5CKzRwVENSaWxXYTZOSDlkCnAzZHhWZ3kyK0Z4cmNqMittMGYwMUN5RjV3U0FWRVVDYWcvUDRBNU1JVVNkTWtYRFVwWHp5bUpYbUFFQzVWOHgKd0VQakFWdWo1VzFMVXVBS2xlSHAwZy9aQUpWdnkzOWNYcE45aFphSUozcGxMdjVOcWtXZHpvSlQrZ2ZONVl6RQpMTTA4SGRYSGFGa0dlSXpGUDRDeGxLVGNqR25EY1p3cWZlS1ZUUlNybWRsbWlLSDVmYW8rcERGQkJHYTFrcUY0CndHZUJaZEhuSi9HLzBPVHlDUnJVdFNKTFpQL3hCaTR6ZHZkRFU0ei9xYit6Q2s2QVlTSzF6S01JSkZuY3Q2aEUKZDA4WGYvMk1PbE9KUE93THIxZWFQTFFxeHNWQXlsN3k5cXFLMkN4NXQ1SHFnZncwTTBsczZ0VlhGTDBaZis4egpnSTZRQU9DdmxQOE9MSC9zcUZZc1JvdXBWcW5JaC9uaUQ0UThKM1RxSVZUM1pjVWtmTE5mY0ZLV0pCQ1h2SFpGCkxaemhvL0paMzhCdkhZS2UzWmdhcUZuUG1rTUhSbW9XR3N2VDEwUmdRNnZyM21QeTNuYVY3WXk1b0JtOHdYc0cKMjd1U0ZnZWFYSDF0RDkydFFLNkNLZzNUTmZ4WjM1ekZPbXQ5MUNCcWxYN3pDV1BUOVU4V0M4OVBQaHVFcjZiTwpxUjFjbEdjSlo1VmFOVkdOQ0ZyaFlHeXQrQ1VyYzVqRXNjczI0UkNvMm9oU1BEMWtIOWp2MVFSTmdWK3hQQmhtCjFTS0JIcTNJQUljQTN0dkdOcEhGSFpMTjlJczBCY0JJQXd2MkhSQlB6S3o0MlpQWGRnVEtQaS9zcFBQbUhwQ0IKNS8zVHd4QlBuUEErdVBES0IweHVnUUtDQVFFQTY2cWdkWC9DVjhJYkd0cmtHdzlkS0V6Ylpsc3cxbUU2K2paRwp6d3FubnVCMUR3aVJaQzk0Mld4UjBhUERveURPWG93dWtGeEJDYmY3RGtHMjdtbHJyc2JpelJIOG1GajJZM2pGCk55S05Fd0Z5eGxUNnJmNmdyZkYyQ3J4T0doRCsvb05RekNQWFd4OXJLaExhbEFBTlVQei9IQUVJSGFHcWxTL3QKSHRjMDNtWS8vZVlLZ09zbHR6SkwzTm9vdEYzTWFVbnNERGY5ZDlFY01JYmM1TmxYTUdJQVhBa2xsN0c4cSt3VgpSMzkvV0tsZlRMQm4zSkRNN1VZKzNMbEt2cGxkdEk5b21IMkZtSUUyc3NFaUFPSnRFcTRJWExsaFl3YkdFRFBCCldsK3ZGRGNPeWRGN1JRWmZnU3Vmd2lHOUZQYUhUQ2VrY2pNZzN1dk5xVmZ3d2ZGRVZ3S0NBUUVBeWJsNGMzY2wKa0hnVnZLNFZPU3BsdkdOYUxMNmU2SXdYdE1lbHhRVUhRZ0FUQ2ZJUXlqMDV1aHhMVVRRVFd1R1VpRjVmT09DVgpZWWRta0RsTis2bE1uSzBtdGN3Y1Y5YlRHZExYUmVIbWhTcGZvVW8xdjROSkQ3Ukh5ckxMMHhvbFVpSUs3WUNvCmZPU1pnaHBtQ3JHbTkybXQ3Rk54cDlnWjg2UUpYb2tsbHlYUWwweldoMmhzU2phZFd5ZTVqckh3Q1B2ODVlWXAKTCtJQUhZR2ZMcmRNK3l0SWNPOFBaR3FXK290VXAwZEFYRVNLWGVsOUNvRkgzbVVTTm1LTGErcW1NemlYNjlnbApMVTFtKzI2Z1ptNU5aUzBQZUhBVDQ3TnJ2dUtUdmhFOXhZMjV4RWQ1M0poaUdvTnNMMUxTREZtc09hR1JnN3hDCkhob05zR3JnQWpsNEVRS0NBUUVBaHNMWFp0bGI1SnNXck5DbXgrS05QdEpVVTcrcHVhMTM4RVRCVVB1aXhuTkwKL1FERlF5anlUa0ptdHg2U2JPSUJvSWRBeVVRMzNMcmNVVmE2cUxLWDVTWE40NzU1VnhUb01CR0xsTytwQkVrNwp2TFl6Y3ZPOEpqdUJhTG90WHpLQTNrRnhIb3NRUndjOFhKN1Radlp3cHZORlhsb29yZWtZaHA4VlZTRnpGZ29hClpiSDZKNlpQdHVhSUF3NnRZQmFUSVZJb0xRU0hhTkNtOGdqc24rQzR2dTRManBXY2JaWXhjWTVhekx2WHNuZUkKQzI2RXRkZ0hsMWNuODBmeHVmUXhaV0xzU09iZVVrcnhZRjZZZHpFRGRqOFdnUTl3NVJPbWh2VlJoOEJYZVpQUwpYUTN4RTVPRllYUDBzeXBmMlhEd0JWaFY3UWZYc0NIZlJRc0luWjY5bVFLQ0FRRUF2ZFdHbnRLZlRTZFUvV3l0Cm1GQm4wVE9HSnRsMFhjK0lkNTN5U0VpRkhmMUg3WElaanRpelBJdG43WkVlTGNaZnF6MTBTdWcrYTd1cUwwdGUKTWRpT3VsclFOUUZTSXUrTlZaM3U3RnE4UHRFemtXdHV6VDFWUWl3YjY0cjYrVHgvcHAxV3YwNzJRQ0k0dkJEdgp1bHlZamZpcnJkeVE5cTBOTHhWTm8rTll0cmJsWjV6dHZodnMzS0lzengzQko2SVlDN3huc0VvNnRFN3dZTlFjCjR5K29pQzJoVEhZTWZDazZ6NmpURDJDa0ZyQ0JyZ01Pb3hZcWRnV1RjQTZvdWZQbVdBbUlIbkFYL0Z2ODc5eVoKb2d0blZqRmpmNm1wYmZ6cEJvZmZGVnpRVHRzaTJLMlM1cCtZb3U3T25uSklMVXN1SWtZRDFIOFRrWlJmRVloaQpLblcrb1FLQ0FRQVo4ajRwS0xucUNWMEQ2ZVVxSmZ4c0RTRGZReW4vSU1rVHRJVDdpdXkrZlV3Wnh4VXFIT3kyCngraFVyVnA2TEQvMW9FRFhoUTRRQlFIcWlTMVpoaWdyNGRHWm5qdjBSSVlTRGlKY3pwVzI5S1hiWUVPOVEzQUcKY1VUQVIybEVvamVqNVFCY0Vmb0ZPdXA0Skp3Y2tXOHZPZWNLWmxrZXZIRlVFNlhPWElBYUovWDdqNHNheUQ5MApya3NwUlBjK0QrTTZYdm1tYUk0ZkI0aVYzelN6SHNhSGJRMGJTZDRZRHpycjBNOVRndnd6V09NUWM3TzBiTEo1CjdoS01Xb2syQzdQc3ZOSW9PUGUxdVp6UVoyVmR3NUFnTVoxY05yNGhXQnR0ZDU4Ui8zSHNDNnU4TnI2Qzg2R0kKQlV3NEllajVzR2Q3T1pFZG45U0tPa0ZQK0NZSGRQYzgKLS0tLS1FTkQgUFJJVkFURSBLRVktLS0tLQo=
---
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  name: private-ca
  namespace: sandbox
spec:
  ca:
    secretName: private-ca-secret
```
</details>

### CODE REVIEW

File: deployments/epinio.go [hunk 1]
```diff
@@ -133,6 +133,17 @@ func (k Epinio) apply(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI,
 	}
 
 	issuer := options.GetStringNG("tls-issuer")
+	if !internalIssuer(issuer) {
+		found, err := c.ClusterIssuerExists(ctx, issuer)
+		if err != nil {
+			return err
+		}
+		if !found {
+			ui.Exclamation().Msgf("Specified cluster issuer '%s' is missing. Please create it first.", issuer)
```
Comment (jimmykarily) [inline deployments/epinio.go:142]: This is printed too lat and it's not even failing the installation. I used a non-existent issuer and the installation went on as if the issuer was there. I think we should rather verify the existence of the issuer before we even start deploying the deployments. This way the user can fix the problem and then run the install command again without having to uninstall Epinio first.

**Review by jimmykarily — CHANGES_REQUESTED** [review]

File: docs/user/howtos/certificate_issuers.md [hunk 2]
```diff
@@ -45,15 +66,68 @@ spec:
...
+
+### Cluster Issuer for Existing Private CA
+
+According to the instructions from https://cert-manager.io/docs/configuration/ca/, follow these steps:
+
+#### Create Secret With CA Cert and Key
+
+If you don't already have a private CA, use a tool like openssl or easy-rsa to create it.
+
+Don't use this in production:
```
[the remark this answers is not in the document]
Comment (manno) [reply docs/user/howtos/certificate_issuers.md:85]: I'm rewording it and removing the warning. It should already be clear to users that those instructions are not meant as guidance on how to set up a production PKI.

File: docs/user/howtos/certificate_issuers.md [hunk 3]
```diff
@@ -45,15 +66,68 @@ spec:
...
+#### Create ClusterIssuer
+
+Then create the cluster issuer:
+
+```
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: private-ca
+  namespace: sandbox
```
[the remark this answers is not in the document]
Comment (manno) [reply docs/user/howtos/certificate_issuers.md:116]: No idea how that got there, the "**cluster** issuer" a cluster scoped resource, it doesn't have a namespace at all.

[the remark this answers is not in the document]
Comment (jimmykarily) [reply docs/user/howtos/certificate_issuers.md:116]: probably copied from the docs: https://cert-manager.io/docs/configuration/ca/#deployment

File: deployments/epinio.go (hunk 1 above)
Comment (manno) [reply deployments/epinio.go]: I moved the check into a 'PreInstallCheck'

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**